rudder-web/src/main/scala/com/normation/rudder/web/components/DateFormaterService.scala

```diff
@@ -64,12 +64,9 @@ object DateFormaterService {
     appendSuffix(" s", " s")
     .toFormatter()
 
-  def formatPeriod(duration:Duration) : String = {
-    if(duration.getMillis < 1000) "less than 1 s"
-    else periodFormatter.print(duration.toPeriod)
-  }
-
   def getFormatedPeriod(start : DateTime, end : DateTime) : String = {
-    formatPeriod(new Duration(start,end))
+    val duration =  new Duration(start,end)
+    if(duration.getMillis < 1000) "less than 1 s"
+    else periodFormatter.print(duration.toPeriodFrom(start))
```

Comment:
in else do:  <pre>{
   val period = duration.toPeriod(PeriodType.YearMonthDayTime)
  periodFormatter.print(period)
}</pre>